proactor/src/main/java/com/iluwatar/proactor/AsynchronousOperation.java

```diff
@@ -0,0 +1,37 @@
+package com.iluwatar.proactor;
+
+import java.io.NotActiveException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Defines a service that can be executed asynchronously.
+ */
+public class AsynchronousOperation {
+
+  /**
+   *  Logger.
+   */
+  private static final Logger LOGGER = LoggerFactory.getLogger(AsynchronousOperation.class);
+
+  /**
+   * Execute the task.
+   *
+   * @param operation task to do
+   * @return the handler
+   */
+  public Handle execute(String operation) throws NotActiveException, InterruptedException {
+    if ("task1".equals(operation)) {
+      LOGGER.info("task1 started!");
+      Thread.sleep(500);
+      LOGGER.info("task1 finished!");
+      return new Handle("short");
+    } else if ("task2".equals(operation)) {
+      LOGGER.info("task2 started!");
+      Thread.sleep(500);
+      LOGGER.info("task2 finished!");
+      return new Handle("long");
+    }
+    throw new NotActiveException("error");
```

Comment:
We could have a separate class hierarchy for tasks. I'm thinking about something resembling command pattern. That would let us get rid of if-else handling.

Reply:
Short and long tasks both take 500ms to process?

Reply:
Well, I try many patterns to get rid of if-else but it will add too many unrelated codes, and I think we cannot get rid of if-else totally because it will appear in the other place, so I didn't change it.

500->600